### ISSUE TITLE
Opencti (Docker) consuming to much memory

### ISSUE THREAD

**Issue by rafael4ndre** [LuatixHQ/opencti#3413]
## Description

I have an instance of Opencti (Docker) deployed. It was up and running for a few months. Eventually it started consuming to much memory (The VM has 34GB of RAM). 

Acording to what I am able to debug, most of the ram is being consumed by the redis server. I already changed the redis trimming a little. At first to 500000 and then to 300000, and finally to 100000. But the redis server is still consuming all the memory. 

Eventually the server crashes and I neet to reboot. 

## Environment

1. OS: Ubuntu 22.04.2 LTS
2. OpenCTI version: 5.7.6

## Reproducible Steps
Steps to create the smallest reproducible scenario:
1. Deploy opencti with docker 
2. Add misp connectors 
3.Let it run for some time. Maybe three or four weeks

## Screenshots (optional)
For now we have this connectors:
![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/aa31452c-aa55-4ee6-a401-42ebd36f88bd)

And this is the resources right after a reboot: 

![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/73d38ebe-96d2-40fd-add6-6c87924942b0)

**Comment by rafael4ndre** [comment]
On this tutorial (https://www.linkedin.com/pulse/opencti-installation-howto-daniel-bender/) the author says that besides seting the redis trimming it would also be necessary to add "vm.overcommit_memory = 1" to a config file to solve the issue. Tried this on my instace but dosent seem to make a diference. I also deployed a new instace for testing with this tweeks implemented from start to see if I can reproduce the issue. 

As soon as I have any news will post here.

**Comment by iman006** [comment]
I have the same problem with the same amount of RAM and CPU, but unfortunately it seems that there is no solution for this problem. Please share any solution if someone has it :(

**Comment by richard-julien** [comment]
This situation in Redis where it consume too much memory with a limited number of events is generally due to a connector that always add different stix_ids at every run/inject. This situation produces after a long period of time extreme big events (multiple Mo) that pushed in redis stream will consume all the memory.

Can you please check the content of the latest message in Redis stream to verify that my hypothesis is the right one? If its the case please tell us the connector that is responsible of this situation. Thanks

**Comment by rafael4ndre** [comment]
Hi @richard-julien ! 

For sure! 
How can I check the redis stream? There is an specific command for that?

**Comment by rafael4ndre** [comment]
Hello again @richard-julien, 

The attached file was obtained from the url https://Myopencti/stream/live .
Does it help? I can see only heartbeats and deletions. Maybe due to a retention policy configured to exclude older entries. 

I let this live stream run from boot to crash, what take only a few minutes. 

[_live.txt](https://github.com/OpenCTI-Platform/opencti/files/11625625/_live.txt)

**Comment by richard-julien** [comment]
Hi @rafael4ndre ,
Thanks for the extract, i bit surprising you have only delete event but nothing weird.
But looks like you have maybe misconfigured the trimming.

`{"lastEventId":"1685105283051-0","firstEventId":"1677697742309-0","firstEventDate":"2023-03-01T19:09:02.309Z","lastEventDate":"2023-05-26T12:48:03.051Z","streamSize":1599206,"connectionId":"f7267d31-b29d-4017-97c7-d3f35c7b2554"}`

streamSize = 1 599 206 that is way more than the 100 000 you said you have configured.

Please double check the trimming configuration of your opencti configuration.

**Comment by rafael4ndre** [comment]
Yes, after I saw that the 100000 value didnt do much to solve the problem I got it back to 500000. This is the current env variable: 
![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/6ce30947-e66e-403a-8e51-3311c8e062fe)

Setted on Portainer env

**Comment by richard-julien** [comment]
It should be `REDIS__TRIMMING` i think. ( double _ )

**Comment by rafael4ndre** [comment]
Added the env variable with a double _:

![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/d8735916-8c0e-48a5-8e9e-d01aa6133830)

Also disabled the rettention pollice to see if it makes any difference, but it crashes the same way after few minutes. New stream attached. Even with the redis__triming with a double _ configured on the env variables, the stream is getting bigger than the limit. 
Is there a way to clear the stream? 

[stream.txt](https://github.com/OpenCTI-Platform/opencti/files/11626790/stream.txt)

**Comment by rafael4ndre** [comment]
@richard-julien any thoughs on the last message? 

Also, do you think I should just deploy a new instance?

**Comment by richard-julien** [comment]
For me the option should correctly trim the stream to expected size. Are you sure about your configuration? Do you correctly restart your container?

**Comment by rafael4ndre** [comment]
> For me the option should correctly trim the stream to expected size. Are you sure about your configuration? Do you correctly restart your container?

I think so. I re-deployed the config using the update the stack button after adding the redis__trimming option:
![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/74ae6237-cca2-4cea-a083-a9b28ea1cded)

As far as I know, this should apply the changes and restart the container. There is anything else you recommend?

**Comment by rafael4ndre** [comment]
In case anyone having the same issue, I was able to limit the memory usage for redis configuring the docker compose file as such:

![image](https://github.com/OpenCTI-Platform/opencti/assets/16193394/e24579a7-2de3-4888-8e33-a064f8f8b1a2)

The maxmemory flag, as stated in redis documentation is set in bytes; 

The redis_triming (or redis__triming with two _) for some reason doesn't always work in a deploy using portainer.

**Comment by iman006** [comment]
> In case anyone having the same issue, I was able to limit the memory usage for redis configuring the docker compose file as such:
> 
> ![image](https://user-images.githubusercontent.com/16193394/247928754-e24579a7-2de3-4888-8e33-a064f8f8b1a2.png)
> 
> The maxmemory flag, as stated in redis documentation is set in bytes;
> 
> The redis_triming (or redis__triming with two _) for some reason doesn't always work in a deploy using portainer.

Thanks for your help. It works well :D
